src/openMVG/sfm/pipelines/localization/SfM_Localizer.cpp

```diff
@@ -161,7 +155,8 @@ bool SfM_Localizer::RefinePose
   // pose
   sfm_data.poses[0] = pose;
   // intrinsic (the shared_ptr does not take the ownership, will not release the input pointer)
-  sfm_data.intrinsics[0] = std::shared_ptr<cameras::IntrinsicBase>(intrinsics, [](cameras::IntrinsicBase*){});
+  std::shared_ptr<cameras::IntrinsicBase> localIntrinsics(intrinsics->clone());
```

Comment:
ok I thought that when we do BA the values were preserved in case of failure but apparently this is not the case, so the function may return with false and the values of the intrinsics changed due to the BA process and not in a 'good' state. It's better to have a local copy then.